### PR TITLE
Add .whitesource file to activate integration scan

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
We @bbarani already enable the access of [WhiteSource integration](https://whitesource.atlassian.net/wiki/spaces/WD/pages/697696422/WhiteSource+for+GitHub.com) with Github.com for this repo.  However, the automatic PR of `.whitesource` is not created. We asked for the support from WhiteSource side and they suggested we could raise one by ourselves. This PR will also set the WhiteSource integration config mode to `Local` to be using the `whitesource.config`. Dashboards team can modify this configuration on their own to customize it. We are providing the one we had for all repos at this time. 

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/721
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by tmarkley** [comment]
Can we add documentation for WhiteSource in our repo so that maintainers understand how to use it? Or do we need to rely on the WhiteSource documentation on their website?

**Comment by zelinh** [comment]
> Can we add documentation for WhiteSource in our repo so that maintainers understand how to use it? Or do we need to rely on the WhiteSource documentation on their website?

According to @bbarani, we would expect teams to rely on the WhiteSource website for their customized configurations.

**Comment by tmarkley** [comment]
Should repos add any details about our usage of WhiteSource in our [SECURITY](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/SECURITY.md) file?

**Comment by zelinh** [comment]
@bbarani Could you add your comments to this PR?

**Comment by bbarani** [comment]
@tmarkley Please refer to the documentation present [here](https://whitesource.atlassian.net/wiki/spaces/WD/pages/697696422/WhiteSource+for+GitHub.com) to understand more about the Whitesource Github.com integration and config parameters.

You can access the user guide [here](https://whitesource.atlassian.net/wiki/spaces/WD/pages/34111720/WhiteSource+User+Guide)

File: whitesource.config [hunk 1]
```diff
@@ -0,0 +1,372 @@
+###############################################################
+# WhiteSource Unified-Agent configuration file
```
Comment (tmarkley) [inline whitesource.config:2]: Can we add links to the WhiteSource documentation and user guide in this file at least?

Comment (zelinh) [reply whitesource.config:2]: Added some useful document links for this file.

**Review by tmarkley — APPROVED** [review]

**Review by kavilla — APPROVED** [review]
Expectation for us to clean up toe config to specific to our repo? Or should we leave it relatively untouched?

**Comment by zelinh** [comment]
> Expectation for us to clean up toe config to specific to our repo? Or should we leave it relatively untouched?

Yes, it's expected to have this WhiteSource Integration maintained and configuration cleaned up by you. You can refer to the configuration parameter page. The `whitesource.config` I included in this PR is just the unified ones for all of our OpenSearch projects repos.

**Pull request merged**